utils/docker/Dockerfile.centos:7

```diff
@@ -49,26 +49,44 @@ ARG NOBUILD
 # use same UID as host and default value of 1000 if not specified
 ARG UID=1000
 ARG DONT_USE_RPMS=true
+ARG OMPI=unknown
 
 
 # for good measure, clean the metadata
 RUN yum clean metadata
 
 # Install basic tools
 RUN yum -y install epel-release
-RUN yum -y install git gcc gcc-c++ make cmake golang libtool scons boost-devel   \
-                   libuuid-devel openssl-devel libevent-devel libtool-ltdl-devel \
-                   libcmocka libcmocka-devel readline-devel                      \
-                   doxygen pandoc flex patch nasm yasm ninja-build meson         \
-                   CUnit-devel libaio-devel python-pep8 lcov                     \
-                   python clang-analyzer sg3_utils libiscsi-devel                \
-                   numactl-devel doxygen graphviz pylint python2-pygithub        \
-                   python34 python34-pylint python34-PyYAML.x86_64 PyYAML        \
-                   python2-paramiko python34-paramiko valgrind rsync             \
-                   libyaml-devel file ShellCheck yum-plugin-copr
+RUN yum -y install \
+           git gcc gcc-c++ make cmake golang libtool boost-devel   \
+           libuuid-devel openssl-devel libevent-devel              \
+           libtool-ltdl-devel libcmocka-devel readline-devel       \
+           doxygen pandoc flex patch nasm yasm ninja-build meson   \
+           CUnit-devel libaio-devel clang-analyzer sg3_utils       \
+           libiscsi-devel numactl-devel graphviz                   \
+           libyaml-devel                                           \
+           scons pylint file ShellCheck python-pep8 lcov  python   \
+           python2-pygithub yum-plugin-copr
+
+# external deps build/repos
+# sadly this doesn't work
+#RUN yum-config-manager --save --setopt=*_job_daos-stack_job_${ext}_job_master_lastSuccessfulBuild_artifact_artifacts_.gpgcheck=0
+RUN if $DONT_USE_RPMS; then exit 0; fi;                                                                                                                     \
+for ext in openpa libfabric mercury pmix ompi; do                                                                                                                     \
+    yum-config-manager --add-repo=https://build.hpdd.intel.com/job/daos-stack/job/${ext}/job/master/lastSuccessfulBuild/artifact/artifacts/;            \
+    echo "gpgcheck = False" >> /etc/yum.repos.d/build.hpdd.intel.com_job_daos-stack_job_${ext}_job_master_lastSuccessfulBuild_artifact_artifacts_.repo; \
+done
+
+RUN if $DONT_USE_RPMS; then exit 0; fi;                   \
+OMPI=${OMPI} yum -y install openpa-devel libfabric-devel mercury-devel pmix-devel ompi-devel
```

Comment:
I have no idea what this does?

If it's installing from the repo added earlier do we want to use --disable-repo=* --enable-repo=<our one>

Reply:
Possibly.  But I generally find that when you are building your own packages is because either a) it's not available in any other repo or b) you need something newer.

I tend to like to let `yum` work things our for itself given all of the knowledge.  If we have a package that is older than what is in another repo we really should be adding version restrictions to our dependency of it so that `yum` with the whole picture will still do the right thing.